core/jvm/src/main/scala/scalaz/zio/lockfree/LockFreeQueue.scala

```diff
@@ -0,0 +1,19 @@
+package scalaz.zio.lockfree
+
+trait LockFreeQueue[A] {
+  val capacity: Int
+
+  def offer(a: A): Boolean
+
+  def poll(): Option[A]
+
+  def size(): Int
+
+  def enqueuedCount(): Long
+
+  def dequeuedCount(): Long
```

Comment:
What can we do with `enqueuedCount()` and `dequeuedCount()`, which we cannot already do with `size()` and `capacity`?

Reply:
Sure, 
- `capacity` is a constant - how many elements a queue can hold max.
- `size` is how many elements a queue is holding now.
- `enqueued/dequeuedCount` are raw counters, can be used to calculate the rate of enqueueing\dequeueing if the know the time delta.

Example: a queue with `capacity` 8, has `enqueuedCount` 20, `dequeuedCount` 17, and `size` 3.